roles/kubernetes/preinstall/tasks/main.yml

```diff
@@ -178,7 +178,13 @@
 - include: etchosts.yml
   tags: [bootstrap-os, etchosts]
 
-- include: growpart-centos-7.yml
-  when: ansible_distribution in ["CentOS","RedHat"] and
+- name: Check if we are running inside a Azure VM
```

Comment:
I assumed people may provision machines on Azure without providing proper Azure Cloud Provider config. For example, I had to do this because it took some time until I had the proper permissions for my Service Principal Account.

So, the test actually tests if we are running inside Azure and not if the cloud provider was set correctly. If you still have doubts about this solution, I can still change this to check for cloud_provider, as at least for me I don't need to run it without cloud_provider anymore.